pyfr/solvers/base/elements.py

```diff
@@ -211,47 +211,81 @@ def _gen_pnorm_fpts(self):
         self._norm_pnorm_fpts = pnorm_fpts / mag_pnorm_fpts[..., None]
         self._mag_pnorm_fpts = mag_pnorm_fpts
 
-    def _get_jac_eles_at(self, pts):
-        nspts, neles, ndims = self.nspts, self.neles, self.ndims
+    def _get_smats(self, pts, retdets=False):
         npts = len(pts)
+        smats_mpts, djacs_mpts = self._smats_mpts
+
+        # Interpolation matrix to pts
+        M0 = self.basis.mbasis.nodal_basis_at(pts)
+
+        # Interpolate smats
+        smats = np.array([np.dot(M0, smat) for smat in smats_mpts])
+        smats = smats.reshape(self.ndims, npts, self.ndims, -1)
+
+        # Interpolate djacs
+        if retdets:
+            djacs = np.dot(M0, djacs_mpts)
+            return smats, djacs
+        else:
+            return smats
 
-        # Form the Jacobian operator
-        jacop = np.rollaxis(self.basis.sbasis.jac_nodal_basis_at(pts), 2)
+    @lazyprop
+    def _smats_mpts(self):
```

Comment:
Rename to note that we also get the djacs.  Maybe `_smats_djacs_mpts`, but I am not 100%.